src/System.Threading.AccessControl/src/System/Security/AccessControl/MutexSecurity.cs

```diff
@@ -0,0 +1,271 @@
+                propagationFlags,
+                type)
+        {
+        }
+
+        public MutexRights MutexRights
+        {
+            get { return (MutexRights)base.AccessMask; }
+        }
+    }
+
+
+    public sealed class MutexAuditRule : AuditRule
+    {
+        public MutexAuditRule(IdentityReference identity, MutexRights eventRights, AuditFlags flags)
+            : this(identity, (int)eventRights, false, InheritanceFlags.None, PropagationFlags.None, flags)
+        {
+        }
+
+        /*  // Not in the spec
+        public MutexAuditRule(string identity, MutexRights eventRights, AuditFlags flags)
+            : this(new NTAccount(identity), (int) eventRights, false, InheritanceFlags.None, PropagationFlags.None, flags)
+        {
+        }
+        */
+
+        internal MutexAuditRule(IdentityReference identity, int accessMask, bool isInherited, InheritanceFlags inheritanceFlags, PropagationFlags propagationFlags, AuditFlags flags)
+            : base(identity, accessMask, isInherited, inheritanceFlags, propagationFlags, flags)
+        {
+        }
+
+        public MutexRights MutexRights
+        {
+            get { return (MutexRights)base.AccessMask; }
+        }
+    }
+
+
+    public sealed class MutexSecurity : NativeObjectSecurity
+    {
+        public MutexSecurity()
+            : base(true, ResourceType.KernelObject)
+        {
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public MutexSecurity(String name, AccessControlSections includeSections)
+            : base(true, ResourceType.KernelObject, name, includeSections, _HandleErrorCode, null)
+        {
+            // Let the underlying ACL API's demand unmanaged code permission.
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        internal MutexSecurity(SafeWaitHandle handle, AccessControlSections includeSections)
+            : base(true, ResourceType.KernelObject, handle, includeSections, _HandleErrorCode, null)
+        {
+            // Let the underlying ACL API's demand unmanaged code permission.
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        private static Exception _HandleErrorCode(int errorCode, string name, SafeHandle handle, object context)
+        {
+            System.Exception exception = null;
+
+            switch (errorCode)
+            {
+                case Interop.mincore.ERROR_INVALID_NAME:
+                case Interop.mincore.ERROR_INVALID_HANDLE:
+                case Interop.mincore.ERROR_FILE_NOT_FOUND:
+                    if ((name != null) && (name.Length != 0))
+                        exception = new WaitHandleCannotBeOpenedException(SR.Format(SR.WaitHandleCannotBeOpenedException_InvalidHandle, name));
+                    else
+                        exception = new WaitHandleCannotBeOpenedException();
+                    break;
+
+                default:
+                    break;
```

Comment:
Same comments as earlier: why the underscore prefix on the method name, and this default can be removed.